hydro_deploy/core/src/localhost/mod.rs

```diff
@@ -156,19 +156,32 @@ impl LaunchedHost for LaunchedLocalhost {
         id: String,
         binary: Arc<(String, Vec<u8>, PathBuf)>,
         args: &[String],
+        perf: Option<PathBuf>,
     ) -> Result<Arc<RwLock<dyn LaunchedBinary>>> {
-        let mut child = Command::new(&binary.2);
+        let mut command = if perf.is_some() {
+            println!("Profiling binary with perf");
+            let mut tmp = Command::new("perf");
+            tmp.args(["record", "-F", "5", "--call-graph", "dwarf,64000", "-o"])
+                .arg(&perf.unwrap())
```

Comment:
instead of doing unwrap, you can do `if let Some(perf) = perf` and then `perf` will be the inner value inside.